presto-main/src/main/java/com/facebook/presto/operator/scalar/JoniRegexpFunctions.java

```diff
@@ -106,6 +108,54 @@ public static Slice regexpReplace(@SqlType("varchar(x)") Slice source, @SqlType(
         return sliceOutput.slice();
     }
 
+    @Description("replaces substrings matching a regular expression by a unary (lambda) function")
+    @ScalarFunction
+    @SqlType("varchar")
+    public static Slice regexpReplace(@SqlType("varchar") Slice source, @SqlType(JoniRegexpType.NAME) Regex pattern, @SqlType("function(array(varchar), varchar)") UnaryFunctionInterface replaceFunction)
```

Comment:
Since we know the input and output type for the lambda, we can provide a specialized function interface that takes `Block` and returns `Slice`, e.g. 

```
    @FunctionalInterface
    public interface RegexpReplaceLambda
            extends LambdaFunctionInterface
    {
        Slice apply(Block x);
    }
```

See `ArrayFilterFunction` for examples on lambda interface specialization.